planning/obstacle_avoidance_planner/src/node.cpp

```diff
@@ -434,16 +441,30 @@ void ObstacleAvoidancePlanner::insertZeroVelocityOutsideDrivableArea(
         use_footprint_polygon_for_outside_drivable_area_check_);
 
       if (is_outside) {
-        publishVirtualWall(traj_point.pose);
         return i;
       }
     }
     return std::nullopt;
   }();
 
-  if (enable_outside_drivable_area_stop_) {
-    if (first_outside_idx) {
-      for (size_t i = *first_outside_idx; i < optimized_traj_points.size(); ++i) {
+  if (first_outside_idx) {
+    const auto stop_idx = [&]() {
+      const auto dist = tier4_autoware_utils::calcDistance2d(
+        optimized_traj_points.at(0), optimized_traj_points.at(*first_outside_idx));
+      const auto dist_with_margin = dist - vehicle_stop_margin_outside_drivable_area_;
+      const auto first_outside_idx_with_margin =
+        motion_utils::insertStopPoint(dist_with_margin, optimized_traj_points);
+      if (first_outside_idx_with_margin) {
+        return *first_outside_idx_with_margin;
+      } else {
```

Comment:
(Just comment. It's okay not to apply the change.)
This `else` can be removed. This is called early return.

Reply:
I have removed it. Thanks again for the comments!!